src/Microsoft.AspNetCore.Antiforgery/Internal/DefaultAntiforgeryTokenSerializer.cs

```diff
@@ -156,7 +161,14 @@ public string Serialize(AntiforgeryToken token)
 
                 writer.Flush();
                 var stream = serializationContext.Stream;
-                return WebEncoders.Base64UrlEncode(_cryptoSystem.Protect(stream.ToArray()));
+                var bytes = _cryptoSystem.Protect(stream.ToArray());
```

Comment:
Can't avoid the `ToArray()` allocation even after working around the need to `stream.GetBuffer()` for .NET 4.51 and `stream.TryGetBuffer()` otherwise. Like the `byte[]` allocation within `Protect()`, this is fallout from the `byte[]` to `byte[]` `IDataProtector` API.